docs/building-a-node-with-ldk/setting-up-a-channel-manager.md

```diff
@@ -0,0 +1,1276 @@
+let bestHeight = // Insert code to fetch best height
+
+channelManager.asConfirm().bestBlockUpdated(header: bestHeader, height: bestHeight)
+chainMonitor.asConfirm().bestBlockUpdated(header: bestHeader, height: bestHeight)
+
+// Finally, tell LDK that chain sync is complete. This will also spawn several
+// background threads to handle networking and event processing.
+channelManagerConstructor.chainSyncCompleted(persister: channelManagerPersister)
+```
+
+  </template>
+
+</CodeSwitcher>
+
+**Implementation notes:**
+
+There are 2 main options for synchronizing to chain on startup:
+
+**Full Blocks or BIP 157/158**
+
+If you are connecting full blocks or using BIP 157/158, then it is recommended to use
+LDK's `lightning_block_sync` sample module as in the example above: the high-level steps that must be done for both `ChannelManager` and each `ChannelMonitor` are as follows:
+
+1. Get the last blockhash that each object saw.
+   - Receive the latest block hash when through [deserializtion](https://docs.rs/lightning/*/lightning/ln/channelmanager/struct.ChannelManagerReadArgs.html) of the `ChannelManager` via `read()`
+   - Each `ChannelMonitor`'s is in `channel_manager.channel_monitors`, as the 2nd element in each tuple
+2. For each object, if its latest known blockhash has been reorged out of the chain, then disconnect blocks using `channel_manager.as_Listen().block_disconnected(..)` or `channel_monitor.block_disconnected(..)` until you reach the last common ancestor with the main chain.
+3. For each object, reconnect blocks starting from the common ancestor until it gets to your best known chain tip using `channel_manager.as_Listen().block_connected(..)` and/or `channel_monitor.block_connected(..)`.
+4. Call `channel_manager.chain_sync_completed(..)` to complete the initial sync process.
+
+**Esplora**
+
+Alternatively, you can use LDK's `lightning-transaction-sync` crate. This provides utilities for syncing LDK via the transaction-based `Confirm` interface.
+
+**Electrum/Esplora**
+
+Otherwise, you can use LDK's `Confirm` interface directly as in the examples above. The high-level steps are as follows:
+
+1. Tell LDK about relevant confirmed and unconfirmed transactions.
+2. Tell LDK what your best known block header and height is.
+3. Call `channel_manager_constructor.chain_sync_completed(..)` to complete the initial sync process.
+
+**References:** [Rust `Confirm` docs](https://docs.rs/lightning/*/lightning/chain/trait.Confirm.html), [Rust `Listen` docs](https://docs.rs/lightning/*/lightning/chain/trait.Listen.html), [Rust `lightning_block_sync` module docs](https://docs.rs/lightning-block-sync/*/lightning_block_sync/), [Rust `lightning_transaction_sync` module docs](https://docs.rs/lightning-transaction-sync/*/lightning_transaction_sync/)
+
+**Dependencies:** `ChannelManager`, `ChainMonitor`, `ChannelMonitor`s
+
+- If providing providing full blocks or BIP 157/158: set of `ChannelMonitor`s
+- If using Electrum: `ChainMonitor`
```

Comment:
Hmm.. this probably needs to be re-written or can be dropped entirely. Both use `ChainMonitor` which is a set of `ChannelMonitor`s.